### PR TITLE
Automatically retrying with `auth_type=azure-cli` when constructing `workspace_clients` on Azure

### LINKED ISSUE #1430: [FEATURE]: Update documentation to manually override azure auth_type for principal_prefix_access
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem statement

Currently, documentation for "Table migration commands" does not specify what to do when azure-cli has already been installed and authenticated, but auth_type is not being picked up by UCX. A proposed solution was accepted in a previous issue [#1249](https://github.com/databrickslabs/ucx/issues/1249)

### Proposed Solution

Update documentation under "Table migration commands" heading to indicate the following:

If Azure CLI has already been installed and authenticated but you see the following error when running table migration commands:

`14:50:33 ERROR [d.labs.ucx] In order to obtain AAD token, Please run azure cli to authenticate.`

Resolve this in MacOS by running the command with an explicit auth type set: `DATABRICKS_AUTH_TYPE=azure-cli databricks labs ucx ...`. To resolve this issue in Windows, proceed with the following steps:

1. Open `%userprofile%` (the path like C:\Users<user name>)
2. Open `.databrickscfg`
3. Change the profile definition by setting `auth_type=azure-cli`
4. Save, close the file, re-run `databricks labs ucx ...`

### Additional Context

_No response_

### LINKED ISSUE #1574: [BUG]: OAuth not configured or not available: cannot get access token: Error: cache: databricks OAuth is not configured for this host
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

I have configured account level  auth via  databricks auth login --host https://adb-54fdfdf.  but looks like sync-workspace-info  is not working here is the debug info

21:30:50 DEBUG [databricks.sdk] Attempting to configure auth: databricks-cli
21:30:50 DEBUG [databricks.sdk] OAuth not configured or not available: cannot get access token: Error: cache: databricks OAuth is not configured for this host

 databricks labs ucx sync-workspace-info --debug
21:30:49  INFO start pid=99891 version=0.218.1 args="databricks, labs, ucx, sync-workspace-info, --debug"
21:30:49 DEBUG Loading installed version info from: /Users/user/.databricks/labs/ucx/state/version.json pid=99891
21:30:49 DEBUG Loading login configuration from: /Users/user/.databricks/labs/ucx/config/login.json pid=99891
21:30:49 DEBUG Using account-level login profile: accounts pid=99891
21:30:49 DEBUG Loading accounts profile from /Users/user/.databrickscfg pid=99891 sdk=true
21:30:49 DEBUG Resolved login: Config: host=https://accounts.azuredatabricks.net/, account_id=2b0b0274-763c-4895-9b4f-9162bf344c58, profile=accounts, config_file=/Users/user/.databrickscfg pid=99891 sdk=true
21:30:49 DEBUG Passing down environment variables: DATABRICKS_HOST, DATABRICKS_ACCOUNT_ID, DATABRICKS_AUTH_TYPE pid=99891
21:30:49 DEBUG Forwarding subprocess: /Users/user/.databricks/labs/ucx/state/venv/bin/python3 /Users/user/.databricks/labs/ucx/lib/src/databricks/labs/ucx/cli.py {"command":"sync-workspace-info","flags":{"log_level":"debug"},"output_type":""} pid=99891
21:30:49 DEBUG starting: /Users/user/.databricks/labs/ucx/state/venv/bin/python3 /Users/user/.databricks/labs/ucx/lib/src/databricks/labs/ucx/cli.py {"command":"sync-workspace-info","flags":{"log_level":"debug"},"output_type":""} pid=99891
21:30:50 DEBUG [databricks.sdk] Loaded from environment
21:30:50 DEBUG [databricks.sdk] Ignoring pat auth, because databricks-cli is preferred
21:30:50 DEBUG [databricks.sdk] Ignoring basic auth, because databricks-cli is preferred
21:30:50 DEBUG [databricks.sdk] Ignoring metadata-service auth, because databricks-cli is preferred
21:30:50 DEBUG [databricks.sdk] Ignoring oauth-m2m auth, because databricks-cli is preferred
21:30:50 DEBUG [databricks.sdk] Ignoring azure-client-secret auth, because databricks-cli is preferred
21:30:50 DEBUG [databricks.sdk] Ignoring github-oidc-azure auth, because databricks-cli is preferred
21:30:50 DEBUG [databricks.sdk] Ignoring azure-cli auth, because databricks-cli is preferred
21:30:50 DEBUG [databricks.sdk] Ignoring external-browser auth, because databricks-cli is preferred
21:30:50 DEBUG [databricks.sdk] Attempting to configure auth: databricks-cli
21:30:50  INFO [databricks.sdk] Using Databricks CLI authentication
21:30:50  INFO [d.labs.ucx] Account ID: 
21:30:50 DEBUG [databricks.sdk] GET /api/2.0/accounts/2dsdsadsadsadsadsad
< 200 OK
< [
<   {
<     "account_id": "",
<     "azure_workspace_info": {
<       "resource_group": "dbx-rg",
<       "subscription_id": ""
<     },
<     "creation_time": 1709337050316,
<     "deployment_name": "adb-5675576.3",
<     "identity_federation_info": {
<       "enable_identity_federation": true
<     },
<     "location": "eastus",
<     "network_connectivity_config_id": "xxxxxx",
<     "pricing_tier": "PREMIUM",
<     "workspace_id": 123456789,
<     "workspace_name": "azdbx",
<     "workspace_status": "RUNNING",
<     "workspace_status_message": "Workspace is running."
<   },
<   "... (2 additional elements)"
< ]
21:30:50 DEBUG [databricks.sdk] Ignoring pat auth, because databricks-cli is preferred
21:30:50 DEBUG [databricks.sdk] Ignoring basic auth, because databricks-cli is preferred
21:30:50 DEBUG [databricks.sdk] Ignoring metadata-service auth, because databricks-cli is preferred
21:30:50 DEBUG [databricks.sdk] Ignoring oauth-m2m auth, because databricks-cli is preferred
21:30:50 DEBUG [databricks.sdk] Ignoring azure-client-secret auth, because databricks-cli is preferred
21:30:50 DEBUG [databricks.sdk] Ignoring github-oidc-azure auth, because databricks-cli is preferred
21:30:50 DEBUG [databricks.sdk] Ignoring azure-cli auth, because databricks-cli is preferred
21:30:50 DEBUG [databricks.sdk] Ignoring external-browser auth, because databricks-cli is preferred
21:30:50 DEBUG [databricks.sdk] Attempting to configure auth: databricks-cli
21:30:50  INFO [databricks.sdk] Using Databricks CLI authentication
21:30:50 DEBUG [databricks.sdk] Ignoring pat auth, because databricks-cli is preferred
21:30:50 DEBUG [databricks.sdk] Ignoring basic auth, because databricks-cli is preferred
21:30:50 DEBUG [databricks.sdk] Ignoring metadata-service auth, because databricks-cli is preferred
21:30:50 DEBUG [databricks.sdk] Ignoring oauth-m2m auth, because databricks-cli is preferred
21:30:50 DEBUG [databricks.sdk] Ignoring azure-client-secret auth, because databricks-cli is preferred
21:30:50 DEBUG [databricks.sdk] Ignoring github-oidc-azure auth, because databricks-cli is preferred
21:30:50 DEBUG [databricks.sdk] Ignoring azure-cli auth, because databricks-cli is preferred
21:30:50 DEBUG [databricks.sdk] Ignoring external-browser auth, because databricks-cli is preferred
21:30:50 DEBUG [databricks.sdk] Attempting to configure auth: databricks-cli
21:30:50 DEBUG [databricks.sdk] OAuth not configured or not available: cannot get access token: Error: cache: databricks OAuth is not configured for this host
21:30:50 DEBUG [databricks.sdk] Ignoring runtime auth, because databricks-cli is preferred
21:30:50 DEBUG [databricks.sdk] Ignoring google-credentials auth, because databricks-cli is preferred
21:30:50 DEBUG [databricks.sdk] Ignoring google-id auth, because databricks-cli is preferred
21:30:50 ERROR [d.l.ucx.sync-workspace-info] Failed to call sync-workspace-info: Traceback (most recent call last):
  File "/Users/user/.databricks/labs/ucx/state/venv/lib/python3.11/site-packages/databricks/sdk/config.py", line 442, in init_auth
    self._header_factory = self._credentials_provider(self)
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/user/.databricks/labs/ucx/state/venv/lib/python3.11/site-packages/databricks/sdk/credentials_provider.py", line 626, in __call__
    raise ValueError(
ValueError: cannot configure default credentials, please check https://docs.databricks.com/en/dev-tools/auth.html#databricks-client-unified-authentication to configure credentials for your preferred authentication method.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/user/.databricks/labs/ucx/state/venv/lib/python3.11/site-packages/databricks/labs/blueprint/cli.py", line 90, in _route
    cmd.fn(**kwargs)
  File "/Users/user/.databricks/labs/ucx/lib/src/databricks/labs/ucx/cli.py", line 90, in sync_workspace_info
    ctx.account_workspaces.sync_workspace_info()
  File "/Users/user/.databricks/labs/ucx/lib/src/databricks/labs/ucx/account.py", line 65, in sync_workspace_info
    for ws in self.workspace_clients(workspaces):
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/user/.databricks/labs/ucx/lib/src/databricks/labs/ucx/account.py", line 51, in workspace_clients
    ws = self.client_for(workspace)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/user/.databricks/labs/ucx/lib/src/databricks/labs/ucx/account.py", line 40, in client_for
    return self._ac.get_workspace_client(workspace)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/user/.databricks/labs/ucx/state/venv/lib/python3.11/site-packages/databricks/sdk/__init__.py", line 837, in get_workspace_client
    config.init_auth()
  File "/Users/user/.databricks/labs/ucx/state/venv/lib/python3.11/site-packages/databricks/sdk/config.py", line 447, in init_auth
    raise ValueError(f'{self._credentials_provider.auth_type()} auth: {e}') from e
ValueError: databricks-cli auth: cannot configure default credentials, please check https://docs.databricks.com/en/dev-tools/auth.html#databricks-client-unified-authentication to configure credentials for your preferred authentication method.
21:30:50  INFO completed execution pid=99891 exit_code=0

### Expected Behavior

_No response_

### Steps To Reproduce

_No response_

### Cloud

AWS

### Operating System

macOS

### Version

latest via Databricks CLI

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by nkvuong**
## Changes
- Automatically retrying with `auth_type=azure-cli` when constructing `workspace_clients` on Azure
- Resolved TODO items for `AccountWorkspaces`
- Added relevant suggestions in `troubleshooting.md`

### Linked issues

Closes #1574, closes #1430

### Functionality 

- [x] added relevant user documentation

### Tests

- [x] manually tested
- [x] verified on staging environment (screenshot attached)

### CODE REVIEW

**Review by JCZuurmond — APPROVED** [review]
LGTM, two minor comments

File: src/databricks/labs/ucx/account/workspaces.py [hunk 1]
```diff
@@ -72,6 +77,35 @@ def create_account_level_groups(self, prompts: Prompts, workspace_ids: list[int]
...
+        """
+        accessible_workspaces = []
+        for workspace in self._ac.workspaces.list():
+            if self._can_administer(workspace):
+                accessible_workspaces.append(workspace)
+        return accessible_workspaces
+
+    def _can_administer(self, workspace: Workspace) -> bool:
+        try:
+            # check if user is a workspace admin
```
[the remark this answers is not in the document]
Comment (nkvuong) [reply src/databricks/labs/ucx/account/workspaces.py]: done

File: src/databricks/labs/ucx/account/workspaces.py [hunk 2]
```diff
@@ -25,16 +25,21 @@ def _workspaces(self):
             yield workspace
 
     def client_for(self, workspace: Workspace) -> WorkspaceClient:
-        return self._ac.get_workspace_client(workspace)
+        ws = self._ac.get_workspace_client(workspace)
+        try:
+            ws.current_user.me()
+        except (PermissionDenied, NotFound, ValueError):
+            if ws.config.auth_type != "azure-cli":  # retry with azure-cli
```
[the remark this answers is not in the document]
Comment (nkvuong) [reply src/databricks/labs/ucx/account/workspaces.py]: I've rewritten this logic, and tested it locally

**Review by JCZuurmond — APPROVED** [review]
lgtm

File: src/databricks/labs/ucx/account/workspaces.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (nkvuong) [reply src/databricks/labs/ucx/account/workspaces.py]: I've rewritten this logic, and tested it locally

**Pull request merged**